lib/mapper.js

```diff
@@ -100,8 +100,8 @@ function getOrGenerateDevId(facade, payload) {
     if (devId && devId.length > 0) {
         return devId;
     }
-    if (payload.username) {
-        devId = "nodev-segment-" + hash(payload.username);
+    if (payload.user_id) {
+        devId = "nodev-segment-" + hash(payload.user_id);
```

Comment:
I'm thinking that we change this to
devId = "nodev-segment-userid-" + hash(payload.user_id);

This is because we have integrations already sending up a hashed username and we don't want another hash of a user_id matching an existing hashed username and sending up events from a wrong device.